FrameworkSystem/DB/DynamicMonitoringDB.py

```diff
@@ -0,0 +1,171 @@
+    :param dict matchFieldsDict: Dictionary with the fields that the 'match' part of the query should contain
+    :return: Returns the query dictionary to be passed to the elasticsearch API library
+    """
+    query = { 'bool': { 'must': [] } }
+    for key in matchFieldsDict:
+      query[ 'bool' ][ 'must' ].append( { 'match': { key : matchFieldsDict[ key ] } } )
+
+    return query
+
+  def insertLog( self, log ):
+    """
+    Creates a new log entry in ElasticSearch
+    :param dict log: Should be a dictionary containing the logging information to be stored
+    """
+    return self.esDB.addToIndex( '%s_index' % self.docType.lower(), self.docType, log )
+
+  def getLastLog( self, host, component, maxAge = 10 ):
+    """
+    Retrieves the last log for the given component
+    :param str host: The name of the host where the component is installed
+    :param str component: The name of the component in the form system_component
+    :param int maxAge: Tells the method how many days in the past to look for the log. If after reaching maxAge days, no logs have been found, an error will be returned
+    :return: The latest log stored in the database for the given component
+    """
+    date = datetime.datetime.utcnow()
+
+    for i in range( maxAge ):
+      indexName = '%s_index-%s' % ( self.docType.lower(), date.strftime( '%Y-%m-%d' ) )
+      try:
+        if self.esDB.checkIndex( indexName ):
+          result = self.esDB.query( indexName, { 'query': \
+            { 'bool' : { 'must': [ { 'match': { 'host': host } }, { 'match': { 'component': component } } ] } } \
+            , 'sort': { 'timestamp': { 'order': 'desc' } }, 'size': 1 } )
+          if len( result[ 'hits' ][ 'hits' ] ) >= 1:
+            return S_OK( [ result[ 'hits' ][ 'hits' ][0] ] )
+      except Exception, e:
+        return S_ERROR( e )
+
+    return S_ERROR( 'No logs for this component found' )
+
+  def getLogHistory( self, host, component, size = 10 ):
+    """
+    Returns a list with the log history for a given component
+    :param str host: The name of the host where the component is installed
+    :param str component: The name of the component in the form system_component
+    :param int size: Indicates how many entries should be retrieved from the log
+    :return: Up to size entries for the given component from the database
+    """
+    logs = []
+    date = datetime.datetime.utcnow()
+
+    while len( logs ) < size:
+      indexName = '%s_index-%s' % ( self.docType.lower(), date.strftime( '%Y-%m-%d' ) )
+
+      try:
+        if self.esDB.checkIndex( indexName ):
+          result = self.esDB.query( indexName, { 'query': \
+            { 'bool' : { 'must': [ { 'match': { 'host': host } }, { 'match': { 'component': component } } ] } } \
+            , 'sort': { 'timestamp': { 'order': 'desc' } }, 'size': size - len( logs ) } )
+        else:
+          break
+      except Exception, e:
```

Comment:
use 
```
except Exception as e
```
instead of 
```
except Exception, e
```
which is the old syntax (still accepted, though)

Here, and all other places of this module. 

And, possibly, define the exception to catch more precisely.